docs/interpreter_checklist.md

```diff
@@ -0,0 +1,55 @@
+# StableHLO Interpreter Checklist
+
+In this document, we summarize the guidelines while implementing and reviewing
+interpreter for an op. We have intentionally tied a few auxiliary action items,
+related to verifier and type inference, with the idea of making progress in
+those fronts alongside the interpreter implementation.
+
+1. While implementing the interpreter for an op, double check with related
+   implementations (e.g.
+   <!-- markdownlint-disable line-length -->
+   [hlo_interpreter](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/hlo/evaluator)
+   <!-- markdownlint-enable line-length -->
+   and
+   <!-- markdownlint-disable line-length -->
+   [mlir_interpreter](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/mlir_hlo/tools/mlir_interpreter)).
+   <!-- markdownlint-enable line-length -->
+   This action item will allow mutual improvements.
+1. Replace the comments, related to input/output constraints, in
+   the implementations, specifically for verifiers and type inference, with
+   "constraint-labels" (e.g. `Cn` in
+   <!-- markdownlint-disable line-length -->
+   [here](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#constraints-2) )
+   <!-- markdownlint-enable line-length -->
+   used in the
+   <!-- markdownlint-disable line-length -->
+   [specification](https://github.com/openxla/stablehlo/blob/main/docs/spec.md).
+   <!-- markdownlint-enable line-length -->
+   This is to explicitly align all the implementations with the specification:
+   the single source of truth, and help to track gaps in those implementations
+   back to the spec.
+   1. It is OK to have a comment with multiple constraint-labels or to have
+      multiple comments with the same constraint-label. It all depends on how
+      the constraint-label(s) are coded in the implementations.
+   1. In case there is a mismatch between the constraints in the
+      implementation VS those in the specification, make sure there is an
+      open issue reflecting that discrepancy.
+1. Make sure that the there are sufficient tests to cover for the
+   implementations (e.g. verifiers, type inference, interpreter) of the
+   specification. Specifically,
```

Comment:
Are you suggesting that for positive cases we add a  interpreter test, and for negative cases (all kinds of failures) we add/check tests in verifiers/type_inferences test files?

Reply:
Thanks for the input!
I revised the text with additional information. PTAL.